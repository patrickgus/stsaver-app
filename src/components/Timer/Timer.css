.Timer {
  width: 250px;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: space-evenly;
}

.Timer__buttons button {
  margin-right: 10px;
  width: 60px;
}
