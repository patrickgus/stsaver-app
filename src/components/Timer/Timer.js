import React from "react";
import { Link } from "react-router-dom";
import "./Timer.css";

function Timer() {
  return (
    <section className="Timer">
      <h3>00:00:00</h3>
      <div className="Timer__buttons">
        <button>Start</button>
        <button>Stop</button>
        <button>Reset</button>
      </div>
      <Link to={"/activity"}>Back</Link>
    </section>
  );
}

export default Timer;
