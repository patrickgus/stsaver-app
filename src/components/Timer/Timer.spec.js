import React from "react";
import ReactDOM from "react-dom";
import renderer from "react-test-renderer";
import Timer from "./Timer";

describe("Timer component", () => {
  it("renders without crashing", () => {
    const div = document.createElement("div");
    ReactDOM.render(
      <Timer defaultBreakLength={5} defaultSessionLength={20} />,
      div
    );
    ReactDOM.unmountComponentAtNode(div);
  });

  it("renders the timer as expected", () => {
    const tree = renderer
      .create(<Timer defaultBreakLength={5} defaultSessionLength={20} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
