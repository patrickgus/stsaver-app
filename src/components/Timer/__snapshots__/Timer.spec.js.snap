// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Timer component renders the timer as expected 1`] = `
<div
  className="Timer"
>
  <div
    className="settings"
  >
    <div
      className="settings-section"
    >
      <label
        id="break-label"
      >
        Break Length
      </label>
      <div>
        <button
          className=""
          id="break-decrement"
          onClick={[Function]}
        >
          -
        </button>
        <span
          id="break-length"
        >
          5
        </span>
        <button
          className=""
          id="break-increment"
          onClick={[Function]}
        >
          +
        </button>
      </div>
    </div>
    <div
      className="settings-section"
    >
      <label
        id="session-label"
      >
        Session Length
      </label>
      <div>
        <button
          className=""
          id="session-decrement"
          onClick={[Function]}
        >
          -
        </button>
        <span
          id="session-length"
        >
          20
        </span>
        <button
          className=""
          id="session-increment"
          onClick={[Function]}
        >
          +
        </button>
      </div>
    </div>
  </div>
  <div
    className="times"
  >
    <div
      className="times-content"
    >
      <label
        id="timer-label"
      >
        Session
      </label>
      <span
        id="time-left"
      >
        20:00
      </span>
    </div>
  </div>
  <div
    className="controller"
  >
    <button
      className="Button "
      id="start_stop"
      onClick={[Function]}
    >
      Start
    </button>
    <button
      className="Button "
      id="reset"
      onClick={[Function]}
    >
      Reset
    </button>
  </div>
  <audio
    id="beep"
    preload="auto"
    src="alarm_clock.ogg"
  />
</div>
`;
